js/kucoinfutures.js

```diff
@@ -1250,10 +1254,16 @@ module.exports = class kucoinfutures extends kucoin {
     async fetchBalance (params = {}) {
         await this.loadMarkets ();
         // only fetches one balance at a time
-        // by default it will only fetch the BTC balance of the futures account
-        // you can send 'currency' in params to fetch other currencies
-        // fetchBalance ({ 'type': 'future', 'currency': 'USDT' })
-        const response = await this.futuresPrivateGetAccountOverview (params);
+        const request = {};
+        const coin = this.safeString (params, 'coin');
+        const code = this.safeString (params, 'code', this.options['code']);
```

Comment:
In this case we should not really define the default value, so we can remove the 3rd argument, `this.options['code']` from line 1259.

We should only send the code if it is defined in `params` or if it is defined in the `this.options['code']` or if it is defined in `this.options['fetchBalance']['code']`:

```JavaScript
const request = {};
const coin = this.safeString (params, 'coin');
let defaultCode = this.safeString (this.options, 'code');
const fetchBalanceOptions = this.safeValue (this.options, 'fetchBalance', {});
defaultCode = this.safeString (fetchBalanceOptions, 'code', defaultCode);
const code = this.safeString (params, 'code', defaultCode);
if (coin !== undefined) {
    request['currency'] = coin;
} else if (code !== undefined) {
    const currency = this.currency (code);
    request['currency'] = currency['id'];
}
```

And then we can comment out the options above in lines 289-292 for reference. This way we will only send the currency id if the user specified a param override or has set one of the options. Note that ``this.options['fetchBalance']['code']` takes precedence over `this.options['code']`, and params take precedence over options.

Reply:
@kroitor why do we include the options in the method logic but then you say to comment them out in the constructor?

Reply:
Because we want the `request` to remain empty and fallback to the exchange's default whatever it is, unless the user specified a param or has enabled one of the options.